packages/did-client/src/did-client.ts

```diff
@@ -32,104 +23,39 @@ interface VeridaSelfTransactionConfigPart  {
  */
 
 export interface DIDClientConfig {
-    // vda-did resolver - only
-    identifier : string // string value "vda:did:0x12...f" or ethereum address
-    chainName? : string 
-    chainId? : string | number
-
-    // common to vda-did resolver & vda-did self transaction config
-    provider? : Provider
-    rpcUrl? : string
-    registry?: string
-    web3?: any
-}
-
-// Interfaces for saving DID Document
-interface VerificationMethodBase {
-    id: string
-    type: string
-    controller: string
-}
-
-interface Controller extends VerificationMethodBase {
-    blockchainAccountId: string
-}
-
-/**
- * Interface to DIDClient instance
- */
-export interface DIDClient {
-
-    authenticate(
-        veridaPrivateKey: string,
-        callType: CallType,
-        web3Config: VeridaSelfTransactionConfigPart|VeridaMetaTransactionConfig
-    ): void;
-
-    getDid(): string | undefined;
-
-    getPublicKey(): string | undefined;
-
-    save(document: DIDDocument): Promise<boolean>;
-
-    get(did?: string): Promise<DIDDocument | null>
-    
+    network: string             // `testnet` OR `mainnet`
+    connectMode: string         // direct OR gasless @todo: use proper enums
+    rpcUrl : string            // blockchain RPC URI to use
```

Comment:
Before updates, users could create a did-client with one of signer, provider, or rprcUrl.
Less configuration makes it clear but users will lose diversity of creation.

Reply:
Correct. For now, I'm fine to restrict to `rpcUrl` or `gasless`.